### PR TITLE
Addition of RFFT2D and IRFFT2D transforms

### LINKED ISSUE #1656: Add IRFFT2 transform
See https://github.com/lightly-ai/lightly/issues/1646 for information

Please use https://pytorch.org/docs/stable/generated/torch.fft.irfft2.html to implement the transform.

### LINKED ISSUE #1657: Add `RFFT2` (fast fourier) transform using https://pytorch.org/docs/stable/generated/torch.fft.rfft2.html

### PR DESCRIPTION
**Pull request by snehilchatterjee**
**Modified Components:**
- `lightly/transforms/__init__.py`

**New Files:**
- `lightly/transforms/irfft2d_transform.py`
- `lightly/transforms/rfft2d_transform.py`

**Enhancement:**
- Added support for RFFT2D and IRFFT2D transforms.

**Fixes Issue:**
- Closes #1656, Closes #1657

### CODE REVIEW

File: lightly/transforms/rfft2d_transform.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        - Tensor of shape (C, H, W), where C is the number of channels.
+        - PIL.Image can also be passed and will be converted to a Tensor internally.
+
+    Output:
+        - Tensor of shape (C, H, W) in the frequency domain, where C is the number of channels.
+    """
+
+    def __init__(self):
+        self.to_tensor = transforms.ToTensor()
+        self.to_pil = transforms.ToPILImage()
```
Comment (guarin) [inline lightly/transforms/rfft2d_transform.py:25]: We shouldn't add `to_tensor` and `to_pil` here. We can expect that images are already converted to tensors before being passed to `RFFT2D`. I think you can completely remove the `__init__` method.

```suggestion
```

**Review by guarin — CHANGES_REQUESTED** [review]
Thanks, this looks great!

Could you also add tests for each transform? You can add:
* `tests/transforms/test_rfft2d_transform.py`
* `tests/transforms/test_irfft2d_transform.py`

They should look like this:
```
def test():
    transform = RFFT2DTransform()
    image = torch.rand(3, 32, 32)
    output = transform(image)
    assert output.shape == # Put expected shape here
```

Regarding the formatting check, you can run `make run` to format everything.

**Review by guarin — APPROVED** [review]
This looks great, thanks a lot for your contribution!

**Comment by snehilchatterjee** [comment]
Thanks for letting me contribute! Looking forward to contributing more :smiley:

**Pull request merged**